core/services/workflows/v2/engine.go

```diff
@@ -148,32 +150,43 @@ func NewEngine(cfg *EngineConfig) (*Engine, error) {
 	return engine, nil
 }
 
-func (e *Engine) start(_ context.Context) error {
+func (e *Engine) start(ctx context.Context) error {
 	e.cfg.Module.Start()
-	e.srvcEng.Go(e.heartbeatLoop)
-	e.srvcEng.Go(e.init)
-	e.srvcEng.Go(e.handleAllTriggerEvents)
+	ctx = context.WithoutCancel(ctx)
+	ctx = contexts.WithCRE(ctx, contexts.CRE{Owner: e.cfg.WorkflowOwner, Workflow: e.cfg.WorkflowID}) // TODO org?
```

Comment:
We can follow up on adding org later. The current code fails open for missing org, so it doesn't block anything.